### PR TITLE
Don't crash on empty query string parameter name

### LINKED ISSUE #1480: Request with query string '=' returns 500
#### DESCRIPTION

```bash
curl 'http://localhost:5000/people?='
```
returns
```json
{
  "links": {
    "self": "http://localhost:5065/people?="
  },
  "errors": [
    {
      "id": "76871a51-79fc-46ba-848b-f8027f5a03ed",
      "status": "500",
      "title": "An unhandled error occurred while processing this request.",
      "detail": "String cannot be null or empty. (Parameter 'parameterName')"
    }
  ]
}

```

#### STEPS TO REPRODUCE

1. Follow https://www.jsonapi.net/getting-started/step-by-step.html
2. Run `curl 'http://localhost:5000/people?='`

#### EXPECTED BEHAVIOR

A proper 4XX explaining the problem.

#### ACTUAL BEHAVIOR

A 500.

### PR DESCRIPTION
**Pull request by verdie-g**
Closes #1480 

#### QUALITY CHECKLIST
- [X] Changes implemented in code
- [X] Complies with our [contributing guidelines](https://github.com/json-api-dotnet/JsonApiDotNetCore/blob/master/.github/CONTRIBUTING.md)
- [X] Adapted tests
- [ ] Documentation updated

### CODE REVIEW

File: src/JsonApiDotNetCore/QueryStrings/QueryStringReader.cs [hunk 1]
```diff
@@ -38,6 +38,11 @@ public void ReadAll(DisableQueryStringAttribute? disableQueryStringAttribute)
 
         foreach ((string parameterName, StringValues parameterValue) in _queryStringAccessor.Query)
         {
+            if (parameterName.Length == 0)
+            {
+                throw new InvalidQueryException("Empty query string parameter name.", null);
```
Comment (bkoelman) [inline src/JsonApiDotNetCore/QueryStrings/QueryStringReader.cs:43]: I don't see why JADNC should forbid using empty parameters. It just shouldn't crash on them.

Comment (verdie-g) [reply src/JsonApiDotNetCore/QueryStrings/QueryStringReader.cs]: Fair

File: test/JsonApiDotNetCoreTests/IntegrationTests/QueryStrings/QueryStringTests.cs [hunk 2]
```diff
@@ -63,6 +63,24 @@ public async Task Can_use_unknown_query_string_parameter()
         httpResponse.ShouldHaveStatusCode(HttpStatusCode.OK);
     }
 
+    [Theory]
+    [InlineData("")]
+    [InlineData("foo")]
```
Comment (bkoelman) [inline test/JsonApiDotNetCoreTests/IntegrationTests/QueryStrings/QueryStringTests.cs:68]:
```suggestion
    [InlineData("bar")]
```
To match up with the test above it.

File: test/JsonApiDotNetCoreTests/IntegrationTests/QueryStrings/QueryStringTests.cs [hunk 3]
```diff
@@ -63,6 +63,24 @@ public async Task Can_use_unknown_query_string_parameter()
...
+    public async Task Should_ignore_query_parameter_with_empty_name(string parameterValue)
```
Comment (bkoelman) [inline test/JsonApiDotNetCoreTests/IntegrationTests/QueryStrings/QueryStringTests.cs:69]: Looking at the names of existing tests in this file:
- Cannot_use_unknown_query_string_parameter
- Can_use_unknown_query_string_parameter
- Cannot_use_empty_query_string_parameter_value

I'd suggest:
```suggestion
    public async Task Can_use_empty_query_string_parameter_name(string parameterValue)
```

**Review by bkoelman — CHANGES_REQUESTED** [review]
Thanks

**Comment by bkoelman** [comment]
Please update to the latest master.

**Review by bkoelman — APPROVED** [review]
LGTM

**Pull request merged**